### PR TITLE
layers: Add WITHOUT_FORMAT_BIT for storage buffer

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Realized the VU talks of how 

> Any VkImageView or **VkBufferView** being read as a storage image or **storage texel buffer** where the image format field of the OpTypeImage is Unknown ...

So added a path in texel buffer descriptors to all check for proper usage of
`VK_FORMAT_FEATURE_2_STORAGE_READ_WITHOUT_FORMAT_BIT` / `VK_FORMAT_FEATURE_2_STORAGE_WRITE_WITHOUT_FORMAT_BIT` 

cc @djdeath

### CODE REVIEW

**Comment by djdeath** [comment]
Thanks indeed that's missing. I was too focused on images ;) :+1:

**Review by ncesario-lunarg — APPROVED** [review]
LGTM! Was there a particular scenario this was fixing (i.e., is there test case that should be added here)?

**Comment by sfricke-samsung** [comment]
So when fixing the spec Lionel pointed out it says `VkImageView or VkBufferView` and I looked and went "hmmm, there is no coverage for the `VkBufferView` path so added it

again, would add a test when I get a supported device working

**Comment by ncesario-lunarg** [comment]
@sfricke-samsung if you could create a short issue reminding us where the coverage is missing (you can assign it to me), I'm good with merging this.

**Comment by sfricke-samsung** [comment]
@ncesario-lunarg I have a test ready, but it is waiting on the changes to the Device Profile layer in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3740 and figured can just rebase then and add it

> where the coverage is missing

The issue is the VU needs to be check for both `STORAGE_IMAGE` and `STORAGE_TEXEL_BUFFER` but was missing the `STORAGE_TEXEL_BUFFER` part

**Comment by sfricke-samsung** [comment]
@ncesario-lunarg rebased with a test for the storage texel buffer case (if CI fails let me know, most likely test failing on HW that actually support the feature and not using mock ICD like I am)

File: tests/vklayertests_pipeline_shader.cpp [hunk 1]
```diff
@@ -2357,6 +2357,129 @@ TEST_F(VkLayerTest, MissingStorageImageFormatWriteForFormat) {
...
+    buffer_create_info.usage = VK_BUFFER_USAGE_STORAGE_TEXEL_BUFFER_BIT;
+    VkBufferObj buffer;
+    buffer.init(*m_device, buffer_create_info);
+
+    VkBufferViewCreateInfo buff_view_ci = LvlInitStruct<VkBufferViewCreateInfo>();
+    buff_view_ci.buffer = buffer.handle();
+    buff_view_ci.format = format;
+    buff_view_ci.range = VK_WHOLE_SIZE;
+    VkBufferView buffer_view;
+    VkResult err = vk::CreateBufferView(m_device->device(), &buff_view_ci, NULL, &buffer_view);
```
Comment (Tony-LunarG) [inline tests/vklayertests_pipeline_shader.cpp:2456]: This needs to be destroyed by the end of the test

Comment (jeremyg-lunarg) [reply tests/vklayertests_pipeline_shader.cpp:2456]: It would be good to use vk_testing::BufferView to make that happen.

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Pull request merged**